.buildkite/pipeline_cpu_template.py

```diff
@@ -0,0 +1,164 @@
+            "m5d.metal": ["m6i.metal"],
+            "m6i.metal": ["m5d.metal"],
+        },
+        "instances": ["m5d.metal", "m6i.metal", "m6a.metal"],
+    },
+}
+
+
+def group_rdmsr(tests):
+    """
+    Generate a group step with specified parameters for each instance
+    and kernel combination
+    https://buildkite.com/docs/pipelines/group-step
+    """
+    # pdb.set_trace()
+    group_step = group(
+        label=tests[BkStep.LABEL],
+        command=tests[BkStep.COMMAND],
+        instances=tests["instances"],
+        platforms=tests["platforms"],
+    )
+    return [group_step]
+
+
+def group_snapshot_restore(test_step):
+    """
+    Generate a group step with specified parameters for each instance
+    and kernel combination and handle "wait" command between steps
+    https://buildkite.com/docs/pipelines/group-step
+    """
+    groups = []
+
+    groups.append(
+        group(
+            label=test_step["snapshot"][BkStep.LABEL],
+            command=test_step["snapshot"][BkStep.COMMAND],
+            instances=test_step["instances"],
+            platforms=DEFAULT_PLATFORMS,
+            timeout=test_step["snapshot"][BkStep.TIMEOUT],
+            artifacts=test_step["snapshot"][BkStep.ARTIFACTS],
+        )
+    )
+    groups.append("wait")
+    snapshot_restore_combinations = []
+    for dp in DEFAULT_PLATFORMS:
+        for src_instance in test_step["instances"]:
+            for dst_instance in [src_instance] + test_step["cross_instances"].get(
+                src_instance, []
+            ):
+                snapshot_restore_combinations.append(
+                    ((dp, src_instance), (dp, dst_instance))
+                )
+
+    steps = []
+    for combination in snapshot_restore_combinations:
+        (snapshot_os, snapshot_kv), snapshot_instance = combination[0]
+        (restore_os, restore_kv), restore_instance = combination[1]
+        restore_commands = [
+            command.format(instance=snapshot_instance, os=snapshot_os, kv=snapshot_kv)
+            for command in test_step["restore"][BkStep.COMMAND]
+        ]
+        restore_label = test_step["restore"][BkStep.LABEL].format(
+            instance=snapshot_instance,
+            snapshot_os=snapshot_os,
+            snapshot_kv=snapshot_kv,
+            restore_instance=restore_instance,
+            restore_os=restore_os,
+            restore_kv=restore_kv,
+        )
+        steps.append(
+            {
+                BkStep.COMMAND: restore_commands,
+                BkStep.LABEL: restore_label,
+                BkStep.TIMEOUT: test_step["restore"][BkStep.TIMEOUT],
+                "agents": [
+                    f"instance={restore_instance}",
+                    f"kv={restore_kv}",
+                    f"os={restore_os}",
+                ],
+            }
+        )
+
+    groups.append({"group": "📸 restores snapshots", "steps": steps})
+    return groups
+
+
+def main():
+    """
+    Generate group template required trigger pipelines for
```

Comment:
nit: `group template required trigger pipelines` -- this is hard to parse for me...

Reply:
"to" was missing